spec/metanorma/cli/template_spec.rb

```diff
@@ -0,0 +1,38 @@
+require "spec_helper"
+
+RSpec.describe Metanorma::Cli::Template do
+  describe ".add" do
+    context "with valid template source" do
+      it "adds the template to metanorma templates" do
+        name = "iso"
+        source = "https://github.com/metanorma/mn-templates-iso"
+        stub_system_home_directory
+
+        templates = Metanorma::Cli::Template.add(
+          name, source, overwrite: true,
+        )
+
+        expect(templates.last[:name]).to eq(name)
+        expect(templates.last[:source]).to eq(source)
+      end
+    end
+
+    context "with duplicate template" do
+      it "does override the existing template" do
+        name = "iso"
+        source = "https://github.com/metanorma/mn-templates-iso"
+
+        expect do
+          Metanorma::Cli::Template.add(name, source)
+          Metanorma::Cli::Template.add(name, source)
+        end.to raise_error(Metanorma::Cli::Errors::DuplicateTemplateError)
+      end
+    end
+  end
+
+  def stub_system_home_directory
```

Comment:
Use the newly defined `stub_system_home_directory` from `Metanorma::Helper`?

Reply:
Good catch!